src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+
+    if (oaccount)
+        /* Note: the current importer assumes at most 2 splits. This means the second split amount
+         * will be the negative of the the first split amount. We also only set the num field once,
+         * for the first split.
+         */
+        trans_add_split (trans_line->trans, oaccount, book, gnc_numeric_neg(amount), "", omemo);
+
+    return trans_line;
+}
+
+/** Creates a list of transactions from parsed data. Transactions that
+ * could be created from rows are placed in transactions;
+ * rows that fail are placed in error_lines. (Note: there
+ * is no way for this function to "fail," i.e. it only returns 0, so
+ * it may be changed to a void function in the future.)
+ * @param parse_data Data that is being parsed
+ * @param account Account with which transactions are created
+ * @param redo_errors TRUE to convert only error data, FALSE for all data
+ * @return 0 on success, 1 on failure
+ */
+int GncTxImport::parse_to_trans (Account* account,
+                                     bool redo_errors)
+{
+    /* Free error_lines and transactions if they
+     * already exist. */
+    if (!redo_errors) /* If we're redoing errors, we save freeing until the end. */
+    {
+        for (auto orig_line : orig_lines)
+            orig_line.second.clear();
+
+        /* FIXME handle memory leak here!
+         * Existing transactions in the map should probably removed before emptying the map
+         */
+        if (!transactions.empty())
+            transactions.clear();
+    }
+
+    /* compute start and end iterators based on user-set restrictions */
+    auto orig_lines_it = orig_lines.begin();
+    std::advance(orig_lines_it, start_row);
+
+    auto orig_lines_max = orig_lines.begin();
+    if (end_row > orig_lines.size())
+        orig_lines_max = orig_lines.end();
+    else
+        std::advance(orig_lines_max, end_row);
+
+    Account *home_account = NULL;
+    auto odd_line = false;
+    parse_errors = false;
+    for (orig_lines_it, odd_line;
+            orig_lines_it != orig_lines_max;
+            ++orig_lines_it, odd_line = !odd_line)
+    {
+        prop_map_t trans_props;
+
+        /* Skip current line if:
+           1. only looking for lines with error AND no error on current line
+           OR
+           2. looking for all lines AND
+              skip_rows is enabled AND
+              current line is an odd line */
+        if ((redo_errors && orig_lines_it->second.empty()) ||
+           (!redo_errors && skip_rows && odd_line))
+            continue;
+
+        auto line = orig_lines_it->first;
+        GncCsvTransLine* trans_line = NULL;
+
+        /* Convert this import line into a map of transaction/split properties. */
+        auto loop_err = false;
+        auto col_types_it = column_types.cbegin();
+        auto line_it = line.cbegin();
+        for (col_types_it, line_it;
+                col_types_it != column_types.cend(),
```

Comment:
The second part of line is used in the assistant code to display line errors. It's not used in the parser code indeed.